src/main/java/org/openjdk/jextract/impl/HeaderFileBuilder.java

```diff
@@ -589,7 +590,7 @@ private void emitPrimitiveTypedefLayout(String javaName, Type type, Declaration
     }
 
     private String newHolderClassName(String javaName) {
-        String holderClassName = STR."\{javaName}$constants";
+        String holderClassName = javaName;
```

Comment:
Strictly speaking, this change is not necessary - but since we emit nested classes with same name as other members for variadic functions too, I don't really see a reason to append the `$constant` prefix to the nested constant class here.

Reply:
In the end, I've reverted this - I realized that I didn't "clean" before running tests and there were several conflicts of the kind already described. The problem is that it is sometimes common to do this:

```
struct foo { ... } foo;
```

Which then leads to ambiguities. So I reverted that back. (for variadic, there's no such issue).